intera_interface/src/motion_interface/interaction_options.py

```diff
@@ -0,0 +1,244 @@
+        self.set_endpoint_name(endpoint_name)
+        self.set_in_endpoint_frame(in_endpoint_frame)
+        self.set_interaction_control_mode(interaction_control_mode)
+
+    def set_header(self, header = None):
+        if header is None:
+            self._data.header.stamp = rospy.get_rostime()
+            self._data.header.seq = 1
+            self._data.header.frame_id = "base"
+        else:
+            self._data.header = header
+
+    def set_interaction_control_active(self, interaction_active = None):
+        """
+        @param interaction_control_active:
+        --> None:  set it to True by default
+        --> [bool]: copy the element.
+        """
+        if interaction_active is None:
+            self.set_interaction_control_active(True) # default value
+        else:
+            self._data.interaction_control_active = interaction_active
+
+    def set_K_impedance(self, K_impedance = None):
+        """
+        @param K_impedance (Cartesian stiffness):
+        --> None:  populate with vector of default values
+        --> [float]:  copy all elements. Checks length.
+        """
+        if K_impedance is None:
+            self._data.K_impedance = [1300.0, 1300.0, 1300.0, 30.0, 30.0, 30.0]
+        elif len(K_impedance) < self.n_dim_cart:
+            rospy.logerr('The number of elements must be 6!')
+        elif len(K_impedance) == self.n_dim_cart:
+            self._data.K_impedance = deepcopy(K_impedance)
+        else:
+            rospy.logerr('Invalid input to K_impedance!')
+
+    def set_max_impedance(self, max_impedance = None):
+        """
+        @param max_impedance (impedance modulation state):
+        --> None:  populate with vector of default values (True)
+        --> bool:  set each dimension by the input boolean
+        --> [bool]:  copy all elements. Checks length.
+        """
+        if max_impedance is None:
+            self._data.max_impedance =  [True, True, True, True, True, True]
+        elif isinstance(max_impedance[0], bool) and len(max_impedance) == 1:
+            self._data.max_impedance = []
+            for i in range(0, self.n_dim_cart):
+                self._data.max_impedance.append(max_impedance[0])
+        elif len(max_impedance) < self.n_dim_cart and len(max_impedance) > 1:
+            rospy.logerr('The number of elements must be 6 or 1!')
+        elif len(max_impedance) == self.n_dim_cart:
+            self._data.max_impedance = deepcopy(max_impedance)
+        else:
+            rospy.logerr('Invalid input to max_impedance!')
+
+    def set_K_nullspace(self, K_nullspace = None):
+        """
+        @param K_nullspace (nullspace stiffness gains):
+        --> None:  populate with vector of default values
+        --> float:  set each dimension by the input float number
+        --> [float]:  copy all elements. Checks length.
+        """
+        if K_nullspace is None:
+            self._data.K_nullspace = [4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0]
```

Comment:
`self._data.K_nullspace = [4.0]*self.n_dim_cart`, though I think this 0.4 parameter should be defined at the top of the class